### DISCUSSION TITLE
LPC-language

### DISCUSSION THREAD

**Discussion by dworkin** [dworkin/dgd#90] [Announcements]
I have started a new organization on Github called [LPC-language](https://github.com/LPC-language). Currently it comprises the [Asheron's Call](https://github.com/LPC-language/AsheronsFall) emulator (never completed) and various repositories for the (alpha) [signal-server](https://github.com/LPC-language/signal-server) project, a rewrite from scratch of the Signal messenger app server in LPC.

The LPC-language organization is intended for LPC projects that I might collaborate on with others.

**Comment by ChristopherA** [comment]
Where is Hydra now?

**Comment by dworkin** [comment]
[Website](https://www.dworkin.nl/hydra), free 32-bit binaries [here](https://download.dworkin.nl/hydra). To run signal-server, you need cloudlib, put signal-server under `/usr/MsgServer`, plus a bunch of extensions (crypto, libsignal, optionally zlib & jit).  Libsignal expects the checkout of the main extensions to be at `ext`, not `lpc-ext`, in the same toplevel directory.

I don't currently have much time, but I will write documentation about what signal-server can currently do and how to install it (or perhaps a container recipe for that purpose).

**Comment by dworkin** [comment]
The [README](https://github.com/LPC-language/signal-server) now takes you through the steps for installing signal-server, and there are pre-built modules for Hydra i386 [here](https://download.dworkin.nl/hydra/ext) (requires openssl 3).

**Comment by ChristopherA** [comment]
Is is feasible that someone could backport signal-server to an open-source version DGD? Or is it too attached to Hydra features? 

I'm still very involved in the cryptographic security industry (day job), and having a heterogenous open source code for the signal protocol could be powerful. But they are not interested unless it is fully open source.

**Comment by dworkin** [comment]
DGD and Hydra are fully compatible. Everything works on DGD.
